src/Uno.UI.Runtime.Skia.Gtk/Helpers/Theming/GtkSystemThemeHelperExtension.cs

```diff
@@ -29,6 +29,7 @@ private void ObserveSystemTheme()
 			{
 				var settings = Settings.Default;
 				settings.AddNotification(nameof(settings.ApplicationPreferDarkTheme), ApplicationPreferDarkThemeHandler);
+				settings.AddNotification(nameof(settings.ThemeName), ApplicationPreferDarkThemeHandler);
```

Comment:
I think there is a possibility that `Settings.Default.ThemeName` changes but it's still dark, in which case we will incorrectly raise SystemThemeChanged event?

Reply:
Yes, you're right. I will need to check and ignore cases when a light theme is changed to another light theme or dark theme to another dark theme